### PR TITLE
Feature/get process data endpoint

### PR DESCRIPTION
**Pull request by MirelaGeorgiou**
- Created a GET process data endpoint
- The API is to retrieve data from a document DB
- The endpoint does not yet connect to a real database (as one does not exist) but is configured so once the db is created, we set up the conn string environment variables
- Endpoint might need to be refactored if domain changes when front end development commences

### CODE REVIEW

File: mat-process-api.Tests/V1/Controllers/ProcessDataControllerAcceptanceTests.cs [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+            var processDataGateway = new ProcessDataGateway(_dbcontext);
+            var processDataUsecase = new ProcessDataUseCase(processDataGateway);
+            Mock<ILogger<ProcessDataController>> logger = new Mock<ILogger<ProcessDataController>>();
+
+            _processDataController = new ProcessDataController(processDataUsecase, logger.Object);
+        }
+
+        [TestCase("sefwefff")]
+        [TestCase("34252345tff")]
+        [TestCase("324t5gehff")]
```
Comment (erbridge) [inline mat-process-api.Tests/V1/Controllers/ProcessDataControllerAcceptanceTests.cs:55]: Do these refer to something?

File: mat-process-api/V1/Domain/ProcessData.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        public DateTime DateCreated { get; set; }
+        [JsonProperty("dateLastModified")]
+        [BsonElement("dateLastModified")]
+        public DateTime DateLastModified { get; set; }
+        [JsonProperty("dateCompleted")]
+        [BsonElement("dateCompleted")]
+        public DateTime DateCompleted { get; set; }
+        [JsonProperty("dataSchemaVersion")]
+        [BsonElement("dataSchemaVersion")]
+        public int DataSchemaVersion { get; set; }
```
Comment (erbridge) [inline mat-process-api/V1/Domain/ProcessData.cs]: Suggest we call this `ProcessDataSchemaVersion`, as it refers directly to the `ProcessData`. I can imagine a case where we want to have different schemas for the different data blobs.

**Review by Duslerke — APPROVED** [review]

File: mat-process-api/V1/Domain/ProcessData.cs (hunk 2 above)
Comment (MirelaGeorgiou) [reply mat-process-api/V1/Domain/ProcessData.cs]: changed

**Review by erbridge — APPROVED** [review]
I haven't looked at it in detail, but at a high level, looks good to me.

**Review by CJWaszczuk — APPROVED** [review]

**Comment by Duslerke** [comment]
Merging this PR. As there seems to be no reason not to.

**Pull request merged**